lib/src/watch.js

```diff
@@ -98,18 +98,22 @@ const websocketWatcher = model => Async((reject, resolve) => {
 const watch = model => Async((reject, _) => {
   const sources = getSources()
 
-  model.log(watchingDirs(sources))
-
   if (model.useServer) {
     websocketWatcher(model)
       .map(maybeUpdateToWatching)
       .fork(model.log, noop)
   }
 
-  const watcher = chokidar.watch([...sources, ...packageFileNames], {
+  const watchTargets = model.watch
+    ? [model.dir, ...sources, ...packageFileNames]
+    : [...sources, ...packageFileNames]
+
+  model.log(watchingDirs(watchTargets))
+
+  const watcher = chokidar.watch(watchTargets, {
     ignoreInitial: true,
     followSymlinks: false,
-    ignored: ['elm-stuff/generated-code/*', /ElmjutsuDumMyM0DuL3.elm/, /\/.#[^/]+\.elm/]
+    ignored: [new RegExp(model.target), /.git/, /.vscode/, /.idea/, /node_modules/, /elm-stuff/, /ElmjutsuDumMyM0DuL3.elm/, /\/.#[^/]+\.elm/]
```

Comment:
these new folders are "personal" to the developer, especially `.vscode` and `.idea`. For example, I might be using a different IDE that creates a different temp folder for caching, and the watcher will still listen to those files.
A solution would be check if there's a `.gitignore` file, read it, and filter the folders

Reply:
Reading `.gitignore` might lead to other problems, for example when you are generating ignored source files that should still trigger a reload.

We could just force `--dir` when using `--watch` or are there use cases where you have to watch the current working directory?

Reply:
that's true. Forcing the `dist` might help if you target another directory other than root, but if you still target the root then the problem will persist.
Maybe we can say "any path (file or folder) that starts with `.` + `/node_modules/` etc, and a flag can be added, like `--watchIgnore` in case the user wants to ignore additional folders.